### ISSUE TITLE
Port developement: vcpkg x-add-version

### ISSUE THREAD

**Issue by jvbsl** [microsoft/vcpkg#33574]
It seems there is a difference with "vcpkg x-add-version ..." on a branch with a remote to your own fork, or if you are not executing it on the master branch.
Don't know exactly which is the problem, but for me it produces the wrong result If I'm not executing it directly on my master branch, which has microsoft/vcpkg set as remote. Would be good if that would be documented somewhere in an understandable way

I could test out which one of the above produces the problem, but would need to create a PR for that to test on

**Comment by dg0yt** [comment]
Please provide steps to reproduce the problem. Many contributors use this command without a problem, on their personal forks and branches. (And basically it doesn't care about remotes.)

**Comment by jvbsl** [comment]
Alright I can't reproduce it for now :D So it probably has nothing to do with the remotes, I have know idea what caused that problem. I already had it two times on two different PRs, I have no idea what exactly I did differently from everyone else. Or even what I do differently from myself when it works. I thought that the branch was the only difference.
So perhaps close it for now, till I have the problem again and then can find out how to reproduce it?

**Comment by dg0yt** [comment]
Remember that you must `git commit` changes to  `ports/<PORT>` before running `vcpkg x-add-version <PORT>`.
(You can amend the previous commit. But you must preserve this order also when amending.)

**Comment by jvbsl** [comment]
Yes I know, that's also what I have done. If I don't commit `vcpkg x-add-version` would do nothing. My problem was that the `git-tree` id was somehow wrong. But as I said I couldn't reproduce it, so I don't exactly know what I did wrong

**Comment by jimwang118** [comment]
If you are currently unable to reproduce this issue, please close this issue. If the same issue occurs in the future, please open this issue again or resubmit an issue.
